src/irc.py

```diff
@@ -129,52 +127,121 @@ def _install_snap_package(
-        """Generate the PEM file content.
+        self._generate_pem_file_local()
+        self._generate_app_registration_local(matrix, config)
+        self._eval_conf_local(db, matrix, config)
+
+    def _generate_pem_file_local(self) -> None:
+        """Generate the PEM file content."""
+        pem_create_command = [
+            "/bin/bash",
+            "-c",
+            f"[[ -f {IRC_BRIDGE_PEM_FILE_PATH} ]] || "
+            f"openssl genpkey -out {IRC_BRIDGE_PEM_FILE_PATH} "
+            f"-outform PEM -algorithm {IRC_BRIDGE_KEY_ALGO} -pkeyopt {IRC_BRIDGE_KEY_OPTS}",
+        ]
+        logger.info("Creating PEM file for IRC bridge.")
+        subprocess.run(pem_create_command, shell=True, check=True, capture_output=True)  # nosec
+
+    def _generate_app_registration_local(
+        self, matrix: DatasourceMatrix, config: CharmConfig
+    ) -> None:
+        """Generate the content of the app registration file.
 
-        Returns:
-            A string
+        Args:
+            matrix: the matrix configuration
+            config: the charm configuration
         """
-        pass
-
-    def _generate_conf_local(self) -> str:
+        app_reg_create_command = [
+            "/bin/bash",
+            "-c",
+            f"[[ -f {IRC_BRIDGE_REGISTRATION_FILE_PATH} ]] || "
+            f"matrix-appservice-irc -r -f {IRC_BRIDGE_REGISTRATION_FILE_PATH}"
+            f" -u https://{matrix.host}:{IRC_BRIDGE_HEALTH_PORT} "
+            f"-c {IRC_BRIDGE_CONFIG_FILE_PATH} -l {config.bot_nickname}",
+        ]
+        logger.info("Creating an app registration file for IRC bridge.")
+        subprocess.run(
+            app_reg_create_command, shell=True, check=True, capture_output=True
+        )  # nosec
+
+    def _eval_conf_local(
+        self, db: DatasourcePostgreSQL, matrix: DatasourceMatrix, config: CharmConfig
+    ) -> None:
         """Generate the content of the irc configuration file.
 
-        Returns:
-            A string
+        Args:
+            db: the database configuration
+            matrix: the matrix configuration
+            config: the charm configuration
+
+        Raises:
+            KeyError: when encountering a KeyError from the configuration file
         """
-        pass
+        with open(f"{IRC_BRIDGE_CONFIG_FILE_PATH}", "r", encoding="utf-8") as config_file:
+            data = yaml.safe_load(config_file)
+        try:
+            db_conn = data["database"]["connectionString"]
+            if db_conn == "" or db_conn != db.uri:
+                db_conn = data["database"]["connectionString"]
+            data["homeserver"]["url"] = f"https://{matrix.host}"
+            data["ircService"]["ident"] = config.ident_enabled
+            data["ircService"]["permissions"] = {}
+            for admin in config.bridge_admins:
+                data["ircService"]["permissions"][admin] = "admin"
+        except KeyError as e:
+            logger.exception("KeyError: {%s}", e)
+            raise
```

Comment:
I's rather have a customer error being thrown here, something like "raise SynapseConfigurationFileError("some message") from e"

Reply:
Alright, I'll change that.